### PR TITLE
Gradient descent

### PR DESCRIPTION
**Pull request by Matteo294**
The algorithm is not optimized. This is a brief to-do list

1. The learning rate is not optimum
2.  We can use the average value of the derivative of log(psi) instead of the average value of psi'/psi (the exponential disappears)
3. Instead of one function for each expectation value we can make one function that takes as a parameter a pointer to a double function
4. Instead of calculating two expectation values and take the difference, can we calculate the expectation value of the difference? Question for MHJ!

### CODE REVIEW

**Comment by Matteo294** [comment]
Merged
